### DISCUSSION TITLE
Rendering ContentItem to Json

### DISCUSSION THREAD

**Discussion by dannoh** [OrchardCMS/OrchardCore#11067] [Q&A] [answered]
I'm new to the OrchardCore side of an existing project, so I'm not up to speed on all the correct terminology and best practices.

End Goal: I want to render my content item as JSON.

Problem: We are calling the following, and this does not call all the Display Drivers for our widgets.
```cs
var model = await DisplayManager.BuildDisplayAsync(context.Source, UpdateModelAccessor.ModelUpdater)
var item = model.Properties.ContainsKey("ContentItem") ? model.Properties["ContentItem"] as ContentElement : null;
return item?.Content as JObject;
```

Reason: The reason I need to run the DisplayDrivers is that we tweak some property values in there. We see these when we render to HTML, but not when we render to JSON.

I see that currently what happens in our HTML Razor flow is that we render the root level .cshtml and that loops through the children and calls Build and Display on each of those children, which in turn does the same on its children in their cshtml templates.

**Comment by Skrypt** [comment]
You can use this nice tool : https://docs.orchardcore.net/en/latest/docs/reference/modules/Contents/#razor-console-log

**Comment by deanmarcussen** [comment, marked as answer]
Using `DisplayDrivers` produces views, either liquid or razor. The models they produce are generally a `ViewModel`.

They do not modify the source content item json.

So you can't use display drivers to produce json.

To create modified JSON output, you could use a decoupled approach, cloning the json of the content item, and modifying it, as you wish

**Comment by hishamco** [comment]
You could use `json` filter too

**Comment by ns8482e** [comment]
@dannoh for changing value and calculating I’d suggest to use content handler instead of display driver

**Reply by dannoh** [reply]
@deanmarcussen Thanks for setting me straight, that makes perfect sense.
@ns8482e That also sounds like the perfect place to do it, I can make my changes on the way in and out.

Appreciate the quick feedback.
